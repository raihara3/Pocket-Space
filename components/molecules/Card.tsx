import React, { memo } from 'react'
import styled from 'styled-components'
import colors from '../colors'

interface Props {
  title: string
  description?: string
  children?: React.ReactNode
}

const Card: React.FC<Props> = ({
  title,
  description,
  children
}) => {
 return (
    <Box>
      <Title>{title}</Title>
      <Contents>
        {description}
      </Contents>
      {children}
    </Box>
 )
}

const Box = styled.div`
  margin: 10px 0 0;
  padding: 10px 15px;
  background-color: ${colors.black02};
  border: 1px solid rgba(255,255,255,0.12);
  border-radius: 4px;
`

const Title = styled.h2`
  font-size: 20px;
  font-weight: normal;
`

const Contents = styled.div`
  margin: 10px 0;
`

export default memo(Card)
