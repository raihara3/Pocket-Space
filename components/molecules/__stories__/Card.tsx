import React from 'react'
import Card from '../Card'

export const normal = () => {
  return (
    <Card
      title='normal'
      description='This is card item.'
    >
    </Card>
  )
}

export const hasContents = () => {
  return (
    <Card
      title='Has contents'
      description='This is card item.'
    >
      <button>button</button>
    </Card>
  )
}

export default {
  title: 'Card'
}
