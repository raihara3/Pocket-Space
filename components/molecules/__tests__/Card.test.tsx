import React from 'react'
import renderer from 'react-test-renderer'
import Card from '../Card'

test('normal', () => {
  const tree = renderer
    .create(
      <Card
        title='normal'
        description='This is card item.'
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('has contents', () => {
  const tree = renderer
    .create(
      <Card
        title='Has contents'
        description='This is card item.'
      >
        <button>button</button>
      </Card>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
