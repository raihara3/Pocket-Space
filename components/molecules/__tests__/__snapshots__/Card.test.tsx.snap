// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has contents 1`] = `
<div
  className="Card__Box-sc-1ny7isr-0 fGMiUp"
>
  <h2
    className="Card__Title-sc-1ny7isr-1 lkVTBY"
  >
    Has contents
  </h2>
  <div
    className="Card__Contents-sc-1ny7isr-2 kuYfTj"
  >
    This is card item.
  </div>
  <button>
    button
  </button>
</div>
`;

exports[`normal 1`] = `
<div
  className="Card__Box-sc-1ny7isr-0 fGMiUp"
>
  <h2
    className="Card__Title-sc-1ny7isr-1 lkVTBY"
  >
    normal
  </h2>
  <div
    className="Card__Contents-sc-1ny7isr-2 kuYfTj"
  >
    This is card item.
  </div>
</div>
`;
